src/Adapter/ArrayAdapter.php

```diff
@@ -49,9 +49,11 @@ public function acknowledge(array $messages)
      */
     public function dequeue(MessageFactoryInterface $factory, $limit)
     {
-        $total = null === $limit ? count($this->queue) : $limit;
+        // If there are no messages then LimitIterator should be passed -1 as the count
+        // to avoid an OutOfBoundsException.
+        $count = null === $limit ? -1 : $limit;
```

Comment:
I'd add some brackets for clearer reading:
`$count = (null === $limit) ? -1 : $limit;`